Sources/FluentKit/Query/QueryBuilder.swift

```diff
@@ -625,113 +671,219 @@ public final class QueryBuilder<Model>
+    where Model: FluentKit.Model, Field: FieldRepresentable
+{
+    return .init(lhs, .greaterThan, rhs)
 }
 
-public struct ModelFilter<Model> where Model: FluentKit.Model {
-    static func make<Field>(
+public func < <Model, Field>(lhs: KeyPath<Model, Field>, rhs: KeyPath<Model, Field>) -> ModelFilter<Model>
+    where Model: FluentKit.Model, Field: FieldRepresentable
+{
+    return .init(lhs, .lessThan, rhs)
+}
+
+public func <= <Model, Field>(lhs: KeyPath<Model, Field>, rhs: KeyPath<Model, Field>) -> ModelFilter<Model>
+    where Model: FluentKit.Model, Field: FieldRepresentable
+{
+    return .init(lhs, .lessThanOrEqual, rhs)
+}
+
+// not sure about these operators yet
+/*
+infix operator ~~
+public func ~~ <Model, Field>(lhs: KeyPath<Model, Field>, rhs: [Field.Value]) -> ModelFilter<Model>
+    where Model: FluentKit.Model, Field: FieldRepresentable
+{
+    return .init(lhs, .subset(inverse: false), .array(rhs.map { .bind($0) }))
+}
+
+infix operator !~
+public func !~ <Model, Field>(lhs: KeyPath<Model, Field>, rhs: [Field.Value]) -> ModelFilter<Model>
+    where Model: FluentKit.Model, Field: FieldRepresentable
+{
+    return .init(lhs, .subset(inverse: true), .array(rhs.map { .bind($0) }))
+}*/
+
+
+public func ~= <Model, Field>(lhs: KeyPath<Model, Field>, rhs: KeyPath<Model, Field>) -> ModelFilter<Model>
+    where Model: FluentKit.Model, Field: FieldRepresentable, Field.Value: CustomStringConvertible
+{
+    return .init(lhs, .contains(inverse: false, .suffix), rhs)
+}
+
+public func ~~ <Model, Field>(lhs: KeyPath<Model, Field>, rhs: KeyPath<Model, Field>) -> ModelFilter<Model>
+    where Model: FluentKit.Model, Field: FieldRepresentable, Field.Value: CustomStringConvertible
+{
+    return .init(lhs, .contains(inverse: false, .anywhere), rhs)
+}
+
+public func =~ <Model, Field>(lhs: KeyPath<Model, Field>, rhs: KeyPath<Model, Field>) -> ModelFilter<Model>
+    where Model: FluentKit.Model, Field: FieldRepresentable, Field.Value: CustomStringConvertible
+{
+    return .init(lhs, .contains(inverse: false, .prefix), rhs)
+}
+
+
+public func !~= <Model, Field>(lhs: KeyPath<Model, Field>, rhs: KeyPath<Model, Field>) -> ModelFilter<Model>
+    where Model: FluentKit.Model, Field: FieldRepresentable, Field.Value: CustomStringConvertible
+{
+    return .init(lhs, .contains(inverse: true, .suffix), rhs)
+}
+
+public func !~~ <Model, Field>(lhs: KeyPath<Model, Field>, rhs: KeyPath<Model, Field>) -> ModelFilter<Model>
+    where Model: FluentKit.Model, Field: FieldRepresentable, Field.Value: CustomStringConvertible
+{
+    return .init(lhs, .contains(inverse: true, .anywhere), rhs)
+}
+
+public func !=~ <Model, Field>(lhs: KeyPath<Model, Field>, rhs: KeyPath<Model, Field>) -> ModelFilter<Model>
+    where Model: FluentKit.Model, Field: FieldRepresentable, Field.Value: CustomStringConvertible
+{
+    return .init(lhs, .contains(inverse: true, .prefix), rhs)
+}
+
+
+public enum ModelFilter<Model> where Model: FluentKit.Model {
```

Comment:
I think we might want a separate struct for holding field filters. `ModelFilter` is only generic on a single model, but when you're comparing two fields, you may want to compare them across models. 

What about something like this:
```swift
ModelValueFilter<Model>
ModelFieldFilter<Left, Right>
```